src/ble/BtpEngine.cpp

```diff
@@ -456,7 +456,7 @@ bool BtpEngine::HandleCharacteristicSend(System::PacketBufferHandle data, bool s
 
         mTxBuf    = std::move(data);
         mTxState  = kState_InProgress;
-        mTxLength = mTxBuf->DataLength();
+        mTxLength = static_cast<uint16_t>(mTxBuf->DataLength());
```

Comment:
Must ensure no overrun.

Reply:
For BLE packets, there should be no data loss with a cast to uint16_t.